pkg/pillar/scripts/device-steps.sh

```diff
@@ -29,6 +29,16 @@ export LOG_TO_SYSLOG
 echo "$(date -Ins -u) Starting device-steps.sh"
 echo "$(date -Ins -u) EVE version: $(cat $BINDIR/versioninfo)"
 
+# wait for rsyslogd to be available for loggin
```

Comment:
I don't think we want this part. We can make the agents simpler if they all log to one place (memlogd for now; perhaps the kernel later) so they don't have to use the logrus syslog plugin. 
I wonder whether the limited 10 second wait in the agents (before they switch to stdout) is sufficient for now as you move it to a service; it is first in the list of services.

Reply:
It might, it might not. Do we want 10 seconds and then switch? Or do we want them to keep trying whatever they are configured for ad infinitum, and let the timeout or other be in watchdog, where policy is managed?

> We can make the agents simpler if they all log to one place (memlogd for now; perhaps the kernel later) so they don't have to use the logrus syslog plugin.

Oh, yes, but you already knew I liked that. :-)

Reply:
@eriknordmark Unlike onboot section containers in service section can keep running in parallel. There is no guarantee rsyslogd would have initialized fully before pillar container starts. At present we use logread for reading logs from memlogd and punt to rsyslogd using logger. If logread starts before rsyslogd is fully initialized, the initial set of onboot logs will be lost.
@rvs has started looking into implementing a containerd plugin that can replace what logread/logger does. Once we have the plugin available this particular wait can be removed.
I'll put a 10 seconds max ceiling for the wait in this loop. I'm currently working on making pillar services directly log to stdout.

Reply:
With the new architecture we've discussed with @gkodali-zededa and @nordmark -- this section in the device-steps is simply not needed.

From now on we are *always* assuming that rsyslog may go off-line for periods of time -- and that's just fine. Hence we don't have to wait for it to actually be online before proceeding here.

For as long as rsyslog is off-line -- logging from anything that goes in the pillar container will go into memlogd and will be scooped up from there.

In fact, I'm actually thinking that direct to syslog logging from services is redundant anyway (it is a tiny optimization -- but its not like it is absolutely needed) but I'm fine with leaving it be for now.

This section, however, needs to be removed.

Reply:
Removed this section